apps/web/src/app/(networks)/(non-evm)/stellar/pool/add/page.tsx

```diff
@@ -293,8 +296,41 @@ export default function AddPoolPage() {
     hasValidAmounts &&
     !isAboveRange &&
     !needsAnyTrustline
-  const isCreating =
-    createAndInitializePoolMutation.isPending || addLiquidityMutation.isPending
+
+  const ctaButtonText = useMemo((): string => {
+    switch (createPoolState) {
+      case 'idle': {
+        if (needsAnyTrustline) {
+          return 'Create trustline first'
+        }
+        if (isAboveRange) {
+          return 'Price Above Range'
+        }
+        if (!token0 || !token1) {
+          return 'Select Tokens'
+        }
+        if (!hasValidAmounts) {
+          return 'Enter Liquidity Amounts'
+        }
+        if (!ticksAligned) {
+          return 'Align Ticks & Continue'
+        }
+        return 'Continue'
+      }
+      case 'adding_liquidity':
+        return 'Adding Liquidity...'
+      case 'creating':
+        return 'Creating/Initializing Pool...'
+    }
+  }, [
+    createPoolState,
+    needsAnyTrustline,
+    isAboveRange,
+    token0,
+    token1,
+    hasValidAmounts,
+    ticksAligned,
+  ])
```

Comment:
Refactored this for slightly better readability.

We have built-in exhaustive switch case checking for this function since TypeScript verifies this returns a `string` and not `undefined` (`void`).